mono/metadata/image.c

```diff
@@ -2201,7 +2250,12 @@ mono_image_load_file_for_image (MonoImage *image, int fileidx)
 	} else {
 		int i;
 		/* g_print ("loaded file %s from %s (%p)\n", name, image->name, image->assembly); */
-		res->assembly = image->assembly;
+		if (!assign_assembly_parent_for_netmodule (res, image, error)) {
+			mono_image_unlock (image);
+			mono_image_close (res);
+			return NULL;
+		}
+
 		for (i = 0; i < res->module_count; ++i) {
```

Comment:
Also come to think of it, why DON'T we do this when loading a module as a "file"?